File: fathersDay/Dsa/Day 34 (762. Prime Number of Set Bits in Binary Representation)/Solution.py
class Solution:
    def PrimeSetBits(self, left: int, right: int) -> int:
        def is_prime(n):
            if n < 2:
                return False
            for i in range(2, int(n**0.5) + 1):
                if n % i == 0:
                    return False
            return True
        
        count = 0
        for num in range(left, right + 1):
            set_bits = bin(num).count('1')
            if is_prime(set_bits):
                count += 1
        
        return count
# Example usage:
solution = Solution()
left = 6
right = 10
result = solution.PrimeSetBits(left, right)
print(result)  # Output: 4 (6, 7, 8, 9 have prime number of set bits)

File: fathersDay/Dsa/Day 34 (762. Prime Number of Set Bits in Binary Representation)/Solution.java
class Solution {
    public boolean isPrime(int n) {
        if (n < 2) {
            return false;
        }
        for (int i = 2; i <= Math.sqrt(n); ++i) {
            if (n % i == 0) {
                return false;
            }
        }
        return true;
    }

    public int countSetBits(int n) {
        int count = 0;
        while (n > 0) {
            n &= (n - 1);
            count++;
        }
        return count;
    }

    public int PrimeSetBits(int left, int right) {
        int count = 0;
        for (int num = left; num <= right; ++num) {
            if (isPrime(countSetBits(num))) {
                count++;
            }
        }
        return count;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int left = 6;
        int right = 10;
        int result = solution.PrimeSetBits(left, right);
        System.out.println(result); // Output: 4
    }
}
